website/docs/style/icon/icon-a11y.md

```diff
@@ -6,46 +6,39 @@ tabs: Design('icon'), A11y('icon-a11y'), API('icon-api'), Example('icon-code'),
-|          | `aria-label`      | `div`             | Defines a string value that labels an interactive element. It is a required for buttons without text content.                                                                                                         |
-|          | `aria-labelledby` | `div`             | The `aria-labelledby` attribute identifies the element (or elements) that labels the element it is applied to. It is a required for buttons without text content.                                                     |
-| `button` |                   | `div`, `a`, `svg` | Icon gets `role="button"` automatically, if you add prop `interactive` to it. Identifies the element as a button. Accessible name for the button is defined by the text content of the element or by adding `aria-label`. |
-| `link`   |                   | `img`, `svg`      | Identifies the `img`/`svg` element as a link.                                                                                                                                                                             |
-
-Find live examples in the [A11y style guide](https://a11y-style-guide.com/style-guide/section-media.html#kssref-media-svgs).
+| Attribute             | Usage                                                                                                                                                                                                                     |
+| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `aria-hidden="false"` | Makes the **non-interactive** icon accessible for the assistive technology. Add this attribute to non-interactive icons if they convey meaningful information, such as status, category and so on, that isn't represented by text. |
+| `aria-label` or `aria-labelledby` | Defines an accessible name for the icon. This attribute is required for elements that don't have any text content and aren't hidden from the assistive technology. <br/>For best accessibility, consider adding a visible [Hint](../../components/tooltip/tooltip), which automatically provides an `aria-label` for its trigger, as demonstrated in the [example](./icon-code).  |
 
 ## Considerations for designers
 
-- If an icon has a function in the interface, it should be conveyed to the user through assistive technologies. In this case you should provide an appropriate `aria-label` for it. For example, if `Trash` icon removes a table row in the interface, add `aria-label="Remove row"`.
+- If an icon has a function in the interface and doesn't have any accompanying text, the function should be conveyed to assistive technology users with an appropriate `aria-label` text. For example, if a `Trash` icon removes a table row in the interface, add an `aria-label="Remove row"` attribute.
```

Comment:
I'm a thinking about these two points. They both about adding assisitive label. Maybe we could combine them?

And do we have examples of icons with text for which we don't show a hint?

Reply:
otherwise I'd keep only the recommendation to add a Hint

Reply:
> I just remembered that you were against showing hints for most obvious icons, so I wasn't confident enough to write that you always need a hint
Ahah, it was about Figma and GitHub icons on our documentation website 😁 I guess we shouldn't give the freedom in the recommendations to leave icons without text and without a hint in the product interfaces.

> otherwise I'd keep only the recommendation to add a Hint
Totally support 👍

Reply:
merged 2 points into 1